pkg/services/grafana-apiserver/service.go

```diff
@@ -98,53 +123,55 @@ func (s *service) start(ctx context.Context) error {
 		return err
 	}
 
-	o.RecommendedOptions.SecureServing.BindAddress = net.ParseIP(certgenerator.DefaultAPIServerIp)
-	o.RecommendedOptions.SecureServing.ServerCert.CertKey = options.CertKey{
+	o.SecureServing.BindAddress = net.ParseIP(certgenerator.DefaultAPIServerIp)
+	o.SecureServing.ServerCert.CertKey = options.CertKey{
 		CertFile: certUtil.APIServerCertFile(),
 		KeyFile:  certUtil.APIServerKeyFile(),
 	}
 
-	if err := o.Complete(); err != nil {
-		return err
+	if err := o.Validate(); len(err) > 0 {
+		return err[0]
 	}
 
-	if err := o.Validate(); err != nil {
+	serverConfig := genericapiserver.NewRecommendedConfig(Codecs)
+	err := o.ApplyTo(serverConfig)
+	if err != nil {
 		return err
 	}
 
-	serverConfig, err := o.Config()
+	rootCert, err := certUtil.GetK8sCACert()
 	if err != nil {
 		return err
 	}
 
-	rootCert, err := certUtil.GetK8sCACert()
+	authenticator, err := newAuthenticator(rootCert)
 	if err != nil {
 		return err
 	}
 
-	serverConfig.ExtraConfig.RESTOptionsGetter = filepath.NewRESTOptionsGetter(s.dataPath, unstructured.UnstructuredJSONScheme)
-	serverConfig.GenericConfig.RESTOptionsGetter = filepath.NewRESTOptionsGetter(s.dataPath, grafanaapiserver.Codecs.LegacyCodec(kindsv1.SchemeGroupVersion))
-	serverConfig.GenericConfig.Config.RESTOptionsGetter = filepath.NewRESTOptionsGetter(s.dataPath, grafanaapiserver.Codecs.LegacyCodec(kindsv1.SchemeGroupVersion))
+	serverConfig.Authentication.Authenticator = authenticator
 
-	authenticator, err := newAuthenticator(rootCert)
+	server, err := serverConfig.Complete().New("grafana-apiserver", genericapiserver.NewEmptyDelegate())
 	if err != nil {
 		return err
 	}
 
-	serverConfig.GenericConfig.Authentication.Authenticator = authenticator
+	apiGroupInfo := genericapiserver.NewDefaultAPIGroupInfo(playlistv1.GroupName, Scheme, metav1.ParameterCodec, Codecs)
+	playlistv1storage := map[string]rest.Storage{}
+	playlistv1storage["playlists"] = &playlistv1.Handler{}
 
-	server, err := serverConfig.Complete().New(genericapiserver.NewEmptyDelegate())
-	if err != nil {
+	apiGroupInfo.VersionedResourcesStorageMap["v1"] = playlistv1storage
+	if err := server.InstallAPIGroup(&apiGroupInfo); err != nil {
```

Comment:
@DanCech here's where the `Scheme` is attached to the server (through `apiGroupInfo`)